crates/key-value-azure/src/store.rs

```diff
@@ -318,55 +342,90 @@ impl AzureCosmosStore {
     async fn get_pair(&self, key: &str) -> Result<Option<Pair>, Error> {
         let query = self
             .client
-            .query_documents(Query::new(format!("SELECT * FROM c WHERE c.id='{}'", key)))
+            .query_documents(Query::new(self.get_query(key)))
             .query_cross_partition(true)
             .max_item_count(1);
 
         // There can be no duplicated keys, so we create the stream and only take the first result.
         let mut stream = query.into_stream::<Pair>();
-        let res = stream.next().await;
-        match res {
-            Some(r) => {
-                let r = r.map_err(log_error)?;
-                match r.results.first().cloned() {
-                    Some((p, _)) => Ok(Some(p)),
-                    None => Ok(None),
-                }
-            }
-            None => Ok(None),
-        }
+        let Some(res) = stream.next().await else {
+            return Ok(None);
+        };
+        Ok(res
+            .map_err(log_error)?
+            .results
+            .first()
+            .map(|(p, _)| p.clone()))
     }
 
     async fn get_keys(&self) -> Result<Vec<String>, Error> {
         let query = self
             .client
-            .query_documents(Query::new("SELECT * FROM c".to_string()))
+            .query_documents(Query::new(self.get_keys_query()))
             .query_cross_partition(true);
         let mut res = Vec::new();
 
         let mut stream = query.into_stream::<Pair>();
         while let Some(resp) = stream.next().await {
             let resp = resp.map_err(log_error)?;
-            for (pair, _) in resp.results {
-                res.push(pair.id);
-            }
+            res.extend(resp.results.into_iter().map(|(pair, _)| pair.id));
         }
 
         Ok(res)
     }
+
+    fn get_query(&self, key: &str) -> String {
+        let mut query = format!("SELECT * FROM c WHERE c.id='{}'", key);
+        self.append_partition_key(&mut query);
+        query
+    }
+
+    fn get_keys_query(&self) -> String {
+        let mut query = "SELECT * FROM c".to_owned();
+        self.append_partition_key(&mut query);
+        query
+    }
+
+    fn get_in_query(&self, keys: Vec<String>) -> String {
+        let in_clause: String = keys
+            .into_iter()
+            .map(|k| format!("'{}'", k))
+            .collect::<Vec<String>>()
+            .join(", ");
+
+        let mut query = format!("SELECT * FROM c WHERE c.id IN ({})", in_clause);
+        self.append_partition_key(&mut query);
+        query
+    }
+
+    fn append_partition_key(&self, query: &mut String) {
+        append_partition_key_condition(query, self.partition_key.as_deref());
+    }
+}
+
+/// Appends an option partition key condition to the query.
+fn append_partition_key_condition(query: &mut String, partition_key: Option<&str>) {
+    if let Some(pk) = partition_key {
+        query.push_str(" AND c.partition_key='");
+        query.push_str(pk);
+        query.push('\'')
```

Comment:
What is the purpose of this trailing `\`?

Reply:
We're pushing single quote (`'`) on to the end of the query (to close the single quote that comes right after the `=`). The `\` escapes the `'` character since characters are always themselves enclosed in `'`'.